lib/rules/no-tracked-properties-from-args.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const ERROR_MESSAGE =
+  'Do not use this.args to create @tracked properties as this will not be updated if the args change. Instead use a getter.';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow creating @tracked properties from this.args',
+      category: 'Ember Decorator',
+      recommended: false,
+      url: 'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-tracked-properties-from-args.md',
+    },
+    fixable: null,
+    schema: [],
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    return {
+      PropertyDefinition(node) {
+        const hasTrackedDecorator =
+          node.decorators?.length > 0 && node.decorators[0].expression.name === 'tracked';
```

Comment:
Can we check against the imported name of `tracked` instead of hardcoding it? It will be a little more robust in case someone does:
```
import { tracked as foobar } from '@glimmer/tracking';
...
@foobar foo = this.args.bar;
```